### PR TITLE
Add S3, gcs and custom connection support (local and workspace)

### PR DESCRIPTION
**Pull request by cheikhgwane**
Add S3, gcs and custom connection support (local and workspace)

### CODE REVIEW

**Comment by qgerome** [comment]
Could you add tests for the connections in the SDK please ?

**Comment by cheikhgwane** [comment]
> Could you add tests for the connections in the SDK please ?

Yes, on it.

File: tests/test_workspace.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        {
+            f"{env_variable_prefix}_HOST": host,
+            f"{env_variable_prefix}_USERNAME": username,
+            f"{env_variable_prefix}_PASSWORD": password,
+            f"{env_variable_prefix}_PORT": port,
+            f"{env_variable_prefix}_DB_NAME": database_name,
+        },
+    ):
+        postgres_connection = workspace.postgresql_connection(slug=slug)
+        assert postgres_connection.url == url
```
Comment (qgerome) [inline tests/test_workspace.py:62]: Could you assert all the fields ?

Comment (cheikhgwane) [reply tests/test_workspace.py:62]: Okay.

**Pull request merged**